test/framework/modules.py

```diff
@@ -178,6 +178,18 @@ def test_load(self):
         for mod in mods:
             self.assertErrorRegex(EasyBuildError, '.*', self.modtool.load, [mod])
 
+        # by default, modules are always loaded, even if they are already loaded
+        self.modtool.load(['GCC/4.7.2'])
+        del os.environ['EBROOTGCC']
+        self.assertTrue('GCC/4.7.2' in self.modtool.loaded_modules())
+        self.modtool.load(['GCC/4.7.2'])
+        self.assertTrue(os.environ.get('EBROOTGCC'))
+
+        # reloading can be disabled using allow_reload=False
+        del os.environ['EBROOTGCC']
+        self.modtool.load(['GCC/4.7.2'], allow_reload=False)
+        self.assertEqual(os.environ.get('EBROOTGCC'), None)
+
```

Comment:
It's not like this will go unnoticed if `$EBROOTGCC` would no longer be set by the GCC module (which is unlikely to happen imho, I see no reason for it?), since we also assert that it *is* set higher up.
We can't check for the order in `$MODULEPATH` with this particular module, since it lives in flat MNS.

I can't think of another way to check whether a module actually gets reloaded... Maybe I can (also) check the `$PATH` entries (we should see the entry for `GCC` added again when it's really reloaded).

When reloading a module, I don't think anything changes in the order in `module list`, even in a module hierarchy?

That said, I can enhance the test if you feel strongly about it, or write a separate test to explicitly check that the order in `$MODULEPATH` in unaffected when modules are being reloaded with `allow_reload=False`, since this test is purely checking what the `load` method does...